diff -Nru jolla/usr/share/jolla-camera/pages/CameraPage.qml alina/usr/share/jolla-camera/pages/CameraPage.qml
--- jolla/usr/share/jolla-camera/pages/CameraPage.qml	2016-01-16 15:55:00.430818000 +0330
+++ alina/usr/share/jolla-camera/pages/CameraPage.qml	2016-01-16 23:09:00.247903470 +0330
@@ -191,5 +191,21 @@
             switcherView.positionViewAtEnd()
             window.activate()
         }
+
+        signal showRearViewfinder()
+        onShowRearViewfinder: {
+            Settings.global.cameraDevice = "primary"
+            Settings.global.captureMode = "image"
+            switcherView.positionViewAtEnd()
+            window.activate()
+        }
+
+        signal showVideoViewfinder()
+        onShowVideoViewfinder: {
+            Settings.global.cameraDevice = "primary"
+            Settings.global.captureMode = "video"
+            switcherView.positionViewAtEnd()
+            window.activate()
+        }
     }
 }
